Monika After Story/game/zz_selector.rpy

```diff
@@ -3319,6 +3467,18 @@ label mas_selector_sidebar_select(items, select_type, preview_selections=True, o
                 for item in items
             ]
 
+        # make filter_map items MASSelectableImageButtonDisplayable
+        if filter_map:
+            filter_map_selbtn = {}
+            for d_item in disp_items:
+                for filter_items in filter_map:
+                    if d_item.selectable in filter_map[filter_items]:
+                        if filter_items in filter_map_selbtn:
+                            filter_map_selbtn[filter_items].append(d_item)
+                        else:
+                            filter_map_selbtn[filter_items] = [d_item]     
```

Comment:
Sorry for the noob question but what do you mean?

Reply:
Right now `filter_map_selbtn` is a `dict`. `defaultdict` accepts a factory as its argument which allows automatically fill missing keys.
E.g. `filter_map_selbtn = collections.defaultdict(list)`. When you access a key that's missing from `filter_map_selbtn`, it will make an empty `list` for that key. So you can just do `filter_map_selbtn["this_key_doesnt_exist"].append(item)` w/o doing that `in` check. [More examples](https://docs.python.org/3/library/collections.html#defaultdict-examples).

Reply:
Thank you I undertsand what you mean now and will implement that

Reply:
This works, tested in my code, will post in the next commit.